### PR TITLE
Feat: Creator Tools nckp format integration

### PR DESCRIPTION
**Pull request by francescosabatini**
- the ksp_compiler.py now improt the Json library and the OrderedDict module
- added a function to parse the performance view JSON file
- added a command into the syntax to import the nckp
- added a cycle to add the parsed ui_controls into the global variables

### CODE REVIEW

File: ksp_compiler3/ksp_compiler.py [hunk 1]
```diff
@@ -1544,13 +1621,19 @@ def do_imports_and_convert_to_line_objects(self):
...
-
-
+                        
         self.lines = parse_lines_and_handle_imports(source,
                                                     read_file_function=self.read_file_func,
                                                     preprocessor_func=self.examine_pragmas)
         handle_conditional_lines(self.lines)
 
+        if import_nckp(source):
+            for line_obj in self.lines:
```
Comment (caux) [inline ksp_compiler3/ksp_compiler.py]: This part could become a `strip_import_nckp_function_from_source` function

File: ksp_compiler3/ksp_compiler.py [hunk 2]
```diff
@@ -1476,6 +1479,80 @@ def compress_variable_name(name):
...
+            key = 'index'
+            for k, v in dictionary.items():
+                if k == key:
+                    cur_prefix.append(prefix[v])
+                    tree = dictionary["value"]["common"]["id"]
+                    yield tree
+                elif isinstance(v, dict):
+                    for result in search_ui_in_dict_recursively(v):
+                        if tree:
+                            name = tree+"_"+result
```
Comment (caux) [inline ksp_compiler3/ksp_compiler.py]: `name = (tree+"_"+result) if tree else result`

Don't know if it's more readable though.

Comment (francescosabatini) [reply ksp_compiler3/ksp_compiler.py]: It is surely more pythonic, I agree

File: ksp_compiler3/ksp_compiler.py (hunk 1 above)
Comment (francescosabatini) [reply ksp_compiler3/ksp_compiler.py]: done

File: ksp_compiler3/ksp_compiler.py (hunk 2 above)
Comment (caux) [reply ksp_compiler3/ksp_compiler.py]: A bit ugly (in my opinion) though. Any other opinions?

Comment (mkruselj) [reply ksp_compiler3/ksp_compiler.py]: The only thing that's maybe ugly in that line is that the concatenation doesn't have extra whitespaces for readability. :)

`name = (tree + "_" + result) if tree else result`

:)

Comment (francescosabatini) [reply ksp_compiler3/ksp_compiler.py]: I can add the extra spaces without problem, but I like the one line solution provided

Comment (mkruselj) [reply ksp_compiler3/ksp_compiler.py]: Me too, it's readable enough.

**Pull request merged**